src/core/src/bound_evaluate.cpp

```diff
@@ -75,85 +75,219 @@ bool is_type_allocable(const element::Type& type) {
+    }
 
-                bool symbols_evaluated = node->evaluate_symbol(output_symbols);
-                for (size_t i = 0; i < outputs.size(); ++i) {
-                    auto& out_tensor = node->get_output_tensor(i);
-                    if (!out_tensor.get_value_symbol().empty())
-                        continue;
-                    if (symbols_evaluated)
-                        out_tensor.set_value_symbol(output_symbols[i]);
-                    if (outputs[i])
-                        ov::populate_tensor_with_missing_symbols(out_tensor);
-                }
+    void set_bound_at_port(descriptor::Tensor& tensor_desc, const size_t port, const bool same_inputs) const {
+        tensor_desc.set_upper_value(bounds[port]);
+        if (same_inputs || are_equal(bounds[port], tensor_desc.get_lower_value())) {
+            tensor_desc.set_lower_value(bounds[port]);
+        }
+    }
+};
 
-                for (const auto& input : input_values) {
-                    auto& tensor = input.get_tensor();
-                    bool should_invalidate = invalidate_all_unused_values;
-                    if (tensor.get_lower_value() && shape_size(tensor.get_lower_value().get_shape()) > 10)
-                        should_invalidate |= true;
+/** @brief Both bounds evaluation specific actions. */
+struct BothBounds {
+    static bool requires_evaluation(const descriptor::Tensor& tensor_desc) {
+        const auto lb = tensor_desc.get_lower_value();
+        const auto ub = tensor_desc.get_upper_value();
+        return static_cast<bool>(lb) != static_cast<bool>(ub) || !lb;
+    }
 
-                    if (tensor.get_upper_value() && shape_size(tensor.get_upper_value().get_shape()) > 10)
-                        should_invalidate |= true;
+protected:
+    void init(const Node& node) {
+        lowers.resize(0);
+        uppers.resize(0);
 
-                    if (should_invalidate && input.get_target_inputs().size() == 1)
-                        tensor.invalidate_values();
-                }
-                propagate_rt_info(node, output);
+        for (const auto& output : node.outputs()) {
+            if (is_type_allocable(output.get_element_type())) {
+                lowers.emplace_back(output);
+                uppers.emplace_back(output);
             } else {
-                break;
+                lowers.emplace_back();
+                uppers.emplace_back();
+            }
+        }
+    }
+
+    void set_bound_at_port(descriptor::Tensor& tensor_desc, const size_t port, const bool same_inputs) const {
+        tensor_desc.set_lower_value(lowers[port]);
+        if (same_inputs || are_equal(lowers[port], uppers[port])) {
+            tensor_desc.set_upper_value(lowers[port]);
+        } else {
+            tensor_desc.set_upper_value(uppers[port]);
+        }
+    }
+
+    bool evaluate(const Node& node) {
+        return node.evaluate_lower(lowers) && node.evaluate_upper(uppers);
+    }
+
+    ov::TensorVector lowers, uppers;
```

Comment:
`ov::TensorVector lowers{}, uppers{};`

Reply:
I shouldn't be required will be default initialized when default constructor will be used.